### PR TITLE
Linux Agent: Clean up handling of GNU 'stat' 5.3 issue (cont'd)

### PR DESCRIPTION
**Pull request by rawiriblundell**
Hi,
continuing on from #22, which I'll close off to simplify my workflow.

The existing section for NFS and CIFS mounts individually deals with a bug in GNU `stat`, resulting in duplicated code.

This commit splits out the detection of the faulty version of `stat` and corrects it for the entire agent, giving consistent `stat` behaviour globally, which allows us to de-duplicate the NFS/CIFS code in the process. This de-duplicated code is tucked away in a new function; `section_nfs_mounts()`

This commit also adds two functions: `get_cifs_mounts()` and `get_nfs_mounts()` which are used by `section_nfs_mounts()` and available for use elsewhere. These provide methods for when `/proc/mounts` exists and when it doesn't.

In other words: the handling for NFS and CIFS from the AIX agent is basically 99% merged in, meaning a step towards merging the various agent scripts.

Further notes/discussion/etc is available for review at #22

### CODE REVIEW

**Comment by rawiriblundell** [comment]
Internal Ref: CMK-2769

**Comment by rawiriblundell** [comment]
Closing due to lack of engagement from tribe29.  In hindsight, I think that a very simple solution to this would be to just broadly override `stat` like this:

```
stat() {
  printf -- '%s\n' "$(command stat "${@}")" | grep .
}
```

`semver_to_int()` has some potential for use cleaning up the handling of openssl and python version checks.
